### ISSUE TITLE
Add date option to the Climatic > Prepare > Climatic Summaries dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5035]
**Is your feature request related to a problem? Please describe.**
There are 3 options (radio buttons at the top) for this dialogue.  This option is only needed for the second button, where the summary is by both year and within year, e.g. month, i.e. Annual + Within.

**Describe the solution you'd like**
A checkbox in this situation labelled **Add Date Column**.
It already adds a Year column, and a Month (say) column.  It combines them to add a date column, which could be called Date.  This is the same as using the Year and Month columns and then going to the Use Date dialogue.  It will make it easier to immediately do time series plots of the resulting data.

For now - at least - the control can take the place of the disabled Omit Unused Levels for this second option.

**Comment by dannyparsons** [comment]
I agree there's no reason we shouldn't do this. I think to make it most general the summary should be "first" date from the date column instead of calculating a date. For example, if you did a within year and used dekad instead of month then it's not obvious how to calculate a date from this but you can always get the first date, which should always be what you want.

This would only not give you the correct answer if the data wasn't ordered by date.

**Comment by dannyparsons** [comment]
If we want to be safer we could have an option for annual summaries to do a calculation of the date from the year as an alternative to first date.

**Comment by dannyparsons** [comment]
@africanmathsinitiative/developers here is another task to be taken up. The suggested is a checkbox on the climatic summaries dialog which produces another summary. This should only be visible when the Annual or Annual + Within buttons are selected.

There still needs to be a discussion on the implementation of the code - someone could start by making the design changes @africanmathsinitiative/developers 

1.

**Comment by dannyparsons** [comment]
Either for @Wycklife or @Ivanluv

**Comment by Muthenya** [comment]
> For now - at least - the control can take the place of the disabled Omit Unused Levels for this second option.
 This is from Roger's initial comment. Should the Omit Missing Values Checkbox be removed?

**Comment by Wycklife** [comment]
@dannyparsons I have already added  the Add date column check  to the climatic summary dialog, and made it visible for annual+within only. I'm awaiting for further discussion on this.

**Comment by Wycklife** [comment]
I'm asking whether the code implementation for this issue has been discussed.

**Comment by dannyparsons** [comment]
It's still not clear to me how we'll do the R code for this. Maybe it would be good to get your pull request merged with the checkbox disabled while the R code is not there.
